tests/5.2/misc/test_printf_in_target_region.c

```diff
@@ -29,8 +29,6 @@ int main () {
    long int long_one = 194834345;
    unsigned long int unsigned_long = 1777444;
    long long int long_long = 282828293898;
-   unsigned char unsigned_char = 200;
```

Comment:
I was thinking about it, but not sure whether it really makes sense.

* Interpreting it as character, i.e. `printf("%c\n", unsigned_char);` – this works, but probably has encoding issues on the terminal as `200` is not a valid UTF-8 character. It would work with 8-bit encodings like ISO 8859-1 (→ "È"), but 8bit / non-UTF-8 encodings are nowadays rather uncommon.

* Using '%u' or '%d' for the output. Well, using `printf("%u\n", unsigned_char);` and `printf("%u\n", unsigned_shortie);` are effectively the same. The C rules convert the variadic argument to "int" in either case – such that 'printf' just sees an int type for both "(un)signed char" and "(un)signed short" (and for "(signed) int").

_(Actually, the latter argument also applies to `%c` as the integer promotion happens in either case.)_

Regarding the latter – more formally (quoting C17 FDIS):
> If an int can represent all values of the original type (as restricted by the width, for a bit-field), the value is converted to an **int**; otherwise, it is converted to an **unsigned int.** These are called the _integer promotions._

and

> If the expression that denotes the called function has a type that does not include a prototype, the integer promotions are performed on each argument ... These are called the _default argument promotions._